pgserver/stmt.go

```diff
@@ -259,3 +261,24 @@ func RemoveComments(query string) string {
 
 	return buf.String()
 }
+
+// precompile a regex to match any "from pg_catalog.pg_stat_replication" in the query.
+var pgCatalogRegex *regexp.Regexp
+
+func getPgCatalogRegex() *regexp.Regexp {
```

Comment:
Not concurrent-safe. You may need `sync.Once*`: https://pkg.go.dev/sync#OnceValue

Reply:
https://pkg.go.dev/regexp#Regexp

> A Regexp is safe for concurrent use by multiple goroutines, except for configuration methods, such as [Regexp.Longest](https://pkg.go.dev/regexp#Regexp.Longest).

So the RWLock is unneeded.